connectors/src/connectors/zendesk/lib/sync_article.ts

```diff
@@ -19,6 +20,36 @@ import type { DataSourceConfig } from "@connectors/types/data_source_config";
 
 const turndownService = new TurndownService();
 
+/**
+ * Deletes an article from the db and the data sources.
+ */
+export async function deleteArticle(
+  connectorId: ModelId,
+  article: ZendeskFetchedArticle,
+  dataSourceConfig: DataSourceConfig,
+  loggerArgs: Record<string, string | number | null>
+): Promise<void> {
+  logger.info(
+    {
+      ...loggerArgs,
+      connectorId,
+      articleId: article.id,
+      name: article.name,
+    },
+    "[Zendesk] Deleting article."
+  );
+  await Promise.all([
```

Comment:
You should do one then the other I think. I would presume you'd want to delete from the data source first and then delete from connectors DB so that it's idempotent?